foundation/foundation-upgrade/src/main/java/com/boozallen/aissemble/upgrade/migration/AbstractPoetryMigration.java

```diff
@@ -0,0 +1,95 @@
+package com.boozallen.aissemble.upgrade.migration;
+
+/*-
+ * #%L
+ * aiSSEMBLE::Foundation::Upgrade
+ * %%
+ * Copyright (C) 2021 Booz Allen
+ * %%
+ * This software package is licensed under the Booz Allen Public License. All Rights Reserved.
+ * #L%
+ */
+
+import java.io.File;
+import java.io.IOException;
+import java.util.Optional;
+import java.util.Set;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.boozallen.aissemble.upgrade.util.FileUtils;
+import com.boozallen.aissemble.upgrade.util.PythonPackageMigration;
+import com.electronwill.nightconfig.core.Config;
+import com.electronwill.nightconfig.core.file.FileConfig;
+
+/**
+ * 
+ * Baton migration used to migrate aiSSEMBLE python packages to the new naming convention and version in pyproject.toml files
+ */
+public abstract class AbstractPoetryMigration extends AbstractAissembleMigration {
+
+    private static final Logger logger = LoggerFactory.getLogger(AbstractPoetryMigration.class);
+    public static final String DEPENDENCIES_KEY = "tool.poetry.dependencies";
+
+    @Override
+    protected boolean shouldExecuteOnFile(File file) {
+        boolean shouldExecute = false;
+        FileConfig poetryConfig = FileConfig.of(file);
+        poetryConfig.load();
+        Optional<Config> dependenciesOpt = poetryConfig.getOptional(DEPENDENCIES_KEY);
+        if(dependenciesOpt.isPresent()) {
+            Config dependencies = dependenciesOpt.get();
+            shouldExecute = hasOldPackages(dependencies);
+        }
+        else {
+            logger.warn("Could not get dependencies for file {}", file.getAbsolutePath());
+        }
+        return shouldExecute;
+    }
+
+    private boolean hasOldPackages(Config currentPackages) {
+        boolean hasOldPackages = false;
+        for(PythonPackageMigration packageMigration : getPythonPackageMigrations()) {
+            if(currentPackages.contains(packageMigration.getOldName())) {
+                hasOldPackages = true;
+                break;
+            }
+        }
+        return hasOldPackages;
+    }
+
+    @Override
+    protected boolean performMigration(File file) {
+        boolean performedSuccessfully = false;
+        FileConfig pyproject = FileConfig.of(file);
```

Comment:
S: does @carter-cundiff's prior comment around just loading the file once still apply here?  I would think it would.

Reply:
After thinking about it some more, I decided it was best not to set an attribute from a file in `shouldExecuteOnFile` and then use that attribute within `performMigration`. Didn't think it was safe to assume the same file would be passed between those two functions consecutively.

Reply:
Okay seems my assumption was wrong. Looking closer at how baton executes, it looks like `shouldExecuteOnFile` and `performMigration` execute on the same file consecutively. So you should be safe to set a `Config config` object in your `shouldExecuteOnFile` and then reference that in your `performMigration`.

Reply:
I'm not sure it would be a good idea to load it once in the shouldExecuteOnFile. Although at the moment it looks like Baton executes the performMigration immediately after the shouldExecuteOnFile, if this ever changed this would not work correctly and would not be obvious.